packages/augur-core/src/contracts/para/WethWrapperForAMMExchange.sol

```diff
@@ -137,4 +137,37 @@ contract WethWrapperForAMMExchange {
         }
         shareToken.unsafeBatchTransferFrom(_from, _to, _tokenIds, _amounts);
     }
+
+    function claimTradingProceeds(IMarket _market, address payable _shareHolder, bytes32 _fingerprint) external {
+        require(shareToken.isApprovedForAll(_shareHolder, address(this)), "Weth wrapper contract must be approved to do share token transfers on behalf of the share holder.");
```

Comment:
I've been letting the share token contract throw an error when lacking approval. It's a bit harder to track though, so either way.